modules/templates/templates_register_api.rb

```diff
@@ -0,0 +1,11 @@
+require 'templates/proxy_request'
+
+class Proxy::TemplatesRegisterApi < Sinatra::Base
+  helpers ::Proxy::Helpers
+
+  get "/" do
+    log_halt(500, "Failed to retrieve Global registration template.") do
+      Proxy::Templates::ProxyRequest.new.get(['register'], request.env, params)
```

Comment:
Btw, I hope you are aware that this sends client certificates from the Smart Proxy. That is an authentication channel. Right now I think you don't have `SmartProxyAuth` in the register controller, but if you do then passing a username and password becomes irrelevant because those client certificates can also authorize. It's quite easy to create a security vulnerability here.

Reply:
> I hope

The worst possible thing when it comes to security :-) I do have the same concern and I think we should enforce HTTP auth. See my comment above.

Reply:
Well, HTTP auth that's then ignored by Foreman isn't really adding a lot.